### DISCUSSION TITLE
Support for Pest 2.0?

### DISCUSSION THREAD

**Discussion by aaronbushnell** [markhuot/craft-pest#92] [Q&A] [answered]
With [Pest 2.0](https://pestphp.com/docs/announcing-pest2) launching today I wanted to see if there was a plan for Craft Pest to upgrade to it?

**Comment by markhuot** [comment, marked as answer]
Yes, indeed! It should be a relatively minor upgrade on the Craft side. I'm planning on giving it a try in the next two weeks.

**Reply by aaronbushnell** [reply]
Awesome! Thanks, @markhuot!

**Reply by bencroker** [reply]
Just checking on this, since I’m using Craft Pest more and more and it would be great to have Pest 2 features available. The only gotcha is that Pest 2 requires PHP ^8.1. Let me know if I can help with testing or anything else!

**Reply by markhuot** [reply]
I have everything working for Craft 3.x but 4.x is currently pinned to a version of `composer/composer` that doesn't play nice with Pest so we need to get `composer/composer` updated upstream first. https://github.com/craftcms/cms/pull/13441

Once that's in I should be able to merge https://github.com/markhuot/craft-pest/pull/98

**Reply by bencroker** [reply]
Fantastic, can’t wait for this and thank you for the amazing work!

**Reply by markhuot** [reply]
Yuck, versions… looks like we might have to wait for Craft 4.5 because of a bad `composer/composer` dep, https://github.com/craftcms/cms/pull/13441#issuecomment-1638851165. I don't think there are any workarounds here… I can't think of a clean way to _force_ this through since `pestphp/pest` and `markhuot/craft-pest` really have no need for `composer/composer`. I _could_ fork it and create an alias but that's a horrible hack…

**Reply by bencroker** [reply]
Yup it does look like it, hopefully it’s not too far off.
